### PR TITLE
Feature connection per request

### PR DESCRIPTION
**Pull request by djeley**

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -21,10 +44,10 @@ sourceSets {
...
 }
 
-repositories 
-{
-    mavenCentral()
-}
+repositories
+        {
+            mavenCentral()
+        }
```
Comment (aaronlong) [inline api/build.gradle]: Should the brackets be in the following format

```
repositories {
    mavenCentral()
}
```

File: api/src/main/java/uk/gov/dvsa/mot/persist/jdbc/MySqlConnectionFactory.java [hunk 2]
```diff
@@ -17,24 +20,57 @@
...
+
+        return ConfigManager.getEnvironmentVariable(ConfigKeys.DatabaseUsername);
+    }
+
+    private String getDbPassword() throws IOException {
+
+        return DB_PASSWORD_HOLDER.getDbPassword();
+    }
+
+    private static class DatabasePasswordHolder {
```
Comment (aaronlong) [inline api/src/main/java/uk/gov/dvsa/mot/persist/jdbc/MySqlConnectionFactory.java]: Would it be better to have this split out into a separate class?  Would make unit testing of this class much easier :)

File: api/src/test/unit/java/uk/gov/dvsa/mot/app/TradeServiceRequestHandlerTest.java [hunk 3]
```diff
@@ -31,10 +31,13 @@
 public class TradeServiceRequestHandlerTest {
     @Mock
     private TradeReadService tradeReadService;
+
     @Mock
     private Context lambdaContext;
+
     private TradeServiceRequest request;
 
+
```
Comment (aaronlong) [inline api/src/test/unit/java/uk/gov/dvsa/mot/app/TradeServiceRequestHandlerTest.java]: extra space here?

File: api/src/main/java/uk/gov/dvsa/mot/app/ConfigManager.java [hunk 4]
```diff
@@ -33,12 +33,26 @@ private ConfigManager() {
...
+     * Retrieve the value of the environment variable with the given name.
+     *
+     * If the code is not running inside a Lambda container, retrieve the same key
+     * from config.properties instead.
+     *
+     * @param name The name of the environment variable to get the value of.
+     * @return The value of the environment variable.
+     * @throws IOException if an error occured getting local config settings
+     */
+    public static String getEnvironmentVariable(String name, boolean decrypt) throws IOException {
```
Comment (cormacq) [inline api/src/main/java/uk/gov/dvsa/mot/app/ConfigManager.java]: I don’t think so, but it’s not a blocker

Comment (cormacq) [inline api/src/main/java/uk/gov/dvsa/mot/app/ConfigManager.java:51]: Could this be a private method?

Comment (djeley) [reply api/src/main/java/uk/gov/dvsa/mot/app/ConfigManager.java:51]: The method is used in DatabasePasswordLoader so needs to be public.

Comment (cormacq) [reply api/src/main/java/uk/gov/dvsa/mot/app/ConfigManager.java:51]: Ah missed that, no problem.

**Review by wkainos — APPROVED** [review]

**Review by cormacq — APPROVED** [review]

**Review by aaronlong — APPROVED** [review]

**Comment by Thomas-Franklin** [comment]
Did no tests need to be updated/added as part of these changes?

**Pull request merged**